### PR TITLE
2023.3.x feature compose blocks auto refresh

### PR DESCRIPTION
**Pull request by EmmyMay**

### CODE REVIEW

**Comment by EmmyMay** [comment]
[Compose page block auto-refresh configurator](https://trello.com/c/I2ToNlOD/3313-compose-page-block-auto-refresh-configurator)

**Comment by EmmyMay** [comment]
[Compose page block auto-refresh configurator](https://trello.com/c/I2ToNlOD/3313-compose-page-block-auto-refresh-configurator)

File: src/compose/types/module-field/select.ts [hunk 1]
```diff
@@ -10,13 +10,15 @@ interface SelectOptions extends Options {
   options: Array<string | { value: string; text?: string }>;
   selectType: string;
   multiDelimiter: string;
+  refreshRate: number;
```
Comment (Fajfa) [inline src/compose/types/module-field/select.ts]: Remove this file

File: src/compose/types/page-block/record-revisions.ts [hunk 2]
```diff
@@ -16,12 +16,14 @@ interface Options {
...
   // referenced fields (records, users) we want to expand
   expRefFields: string[];
+  refreshRate: number;
 }
 
 const defaults: Readonly<Options> = Object.freeze({
   preload: false,
   displayedFields: [],
   expRefFields: [],
+  refreshRate: 0
```
Comment (Fajfa) [inline src/compose/types/page-block/record-revisions.ts]: Lint your code before committing!

**Review by Fajfa — CHANGES_REQUESTED** [review]

File: src/compose/types/page-block/progress.ts [hunk 3]
```diff
@@ -67,6 +70,7 @@ export class PageBlockProgress extends PageBlock {
   }
 
   applyOptions (o?: Partial<Options>): void {
+    Apply(this.options, o, Number, 'refreshRate')
```
Comment (Fajfa) [inline src/compose/types/page-block/progress.ts]: After the !o return line!
Also double check for same problem in other files you changed. 

Apply should be one after the other and right after the !o return line

**Review by Fajfa — APPROVED** [review]

**Pull request closed without merging**